ui/tests/integration/components/sync/secrets/page/destinations-test.js

```diff
@@ -0,0 +1,144 @@
+          @destinations={{this.destinations}}
+          @typeFilter={{this.typeFilter}}
+          @nameFilter={{this.nameFilter}}
+        />`,
+        { owner: this.engine }
+      );
+    };
+
+    this.transitionStub = sinon.stub(this.owner.lookup('service:router'), 'transitionTo');
+    this.clearDatasetStub = sinon.stub(store, 'clearDataset');
+  });
+
+  test('it should render header and tabs', async function (assert) {
+    await this.renderComponent();
+    assert.dom('[data-test-crumb="0"]').includesText('Secrets sync', 'Breadcrumb renders');
+    assert.dom('[data-test-page-title]').hasText('Secrets sync', 'Page title renders');
+    assert.dom('[data-test-tab="Overview"]').exists('Overview tab renders');
+    assert.dom('[data-test-tab="Destinations"]').exists('Destinations tab renders');
+  });
+
+  test('it should render toolbar filters and actions', async function (assert) {
+    assert.expect(4);
+
+    const didTransition = (key, value) => {
+      return this.transitionStub.calledWith('vault.cluster.sync.secrets.destinations', {
+        queryParams: { [key]: value },
+      });
+    };
+
+    this.typeFilter = 'aws-sm';
+    await this.renderComponent();
+
+    assert
+      .dom('[data-test-create-destination]')
+      .includesText('Create new destination', 'Create action renders');
+    assert
+      .dom('[data-test-filter="type"]')
+      .includesText('AWS Secrets Manager', 'Filter is populated for correct initial value');
+    await click('[data-test-selected-list-button="delete"]');
+
+    for (const filter of ['type', 'name']) {
+      await click(`[data-test-filter="${filter}"] .ember-basic-dropdown-trigger`);
+      await click('[data-option-index="0"]');
+
+      const value = filter === 'type' ? 'aws-sm' : 'destination-aws';
+      assert.true(
+        didTransition(filter, value),
+        `${filter} filter triggered transition with correct query params`
+      );
+    }
+  });
+
+  test('it should render empty state when there are no filtered results', async function (assert) {
+    this.destinations = [];
+    this.typeFilter = 'foo';
+    this.nameFilter = 'bar';
+
+    await this.renderComponent();
+
+    const selector = '[data-test-empty-state-title]';
+    assert
+      .dom(selector)
+      .hasText(
+        'There are no foo destinations matching "bar".',
+        'Renders correct empty state when both type and name filters are defined'
+      );
+
+    this.set('nameFilter', undefined);
+    assert
+      .dom(selector)
+      .hasText('There are no foo destinations.', 'Renders correct empty state when type filter is defined');
+
+    this.setProperties({
+      typeFilter: undefined,
+      nameFilter: 'bar',
+    });
+    assert.dom(selector).hasText('There are no destinations matching "bar".');
+  });
+
+  test('it should render destination list items', async function (assert) {
+    // assert.expect(6);
+
+    // this.server.delete('/sync/destinations/aws-sm/destination-aws', () => {
```

Comment:
Do we want to keep this comment?

Reply:
Yes I should have put the TODO at the top but I actually have another PR #23953 where I add the capabilities and uncomment the rest of the this test. It looks like a rogue comment so thanks for pointing it out!